src/common/bitstream.cpp

```diff
@@ -0,0 +1,392 @@
+#include <algorithm>
+#include "bitstream.h"
+#include "util.h"
+
+#define BITS_TO_BYTES(bits) (((bits) + 7) / 8)
+
+
+BitStream::BitStream(std::vector<uint8_t>& exisitingBuf) :
+    buf(exisitingBuf),
+    streamBitPos(0),
+    lastError(Error::NONE) {
+
+}
+
+size_t BitStream::getSizeBits() const {
+    return buf.size() * 8;
+}
+
+size_t BitStream::getRemainingBits() const {
+    size_t sizeBits = getSizeBits();
+    if(streamBitPos >= sizeBits) {
+        return 0;
+    }
+    else {
```

Comment:
I like putting them on newlines to create a visual interrupt between changes in control statement blocks.  If the next part is an `else if`, that also means the conditional doesn't flush vertically with the previous block.  For now, I'll conform.

When I collapse an if...else... statement's block where the closing parenthesis for one block has the beginning of the next block on the same line (like you're suggesting), the collapsed-to statement gets pushed up to immediately follow after the previous statement.  It looks like `if (something) { ... } else if (somethingElse) { ... }` on one line when the if block is collapsed.   Is there a formatting rule that starts the next un-collapsed block on a new line?